### ISSUE TITLE
Fully masked chromosome with Ns gets merged to the previous chromosome 

### ISSUE THREAD

**Issue by ccwang002** [BenLangmead/bowtie2#205]
When we wanted to prevent any reads mapped to chr7 by masking its entire sequence with Ns, chr7 was missing in the BAM file produced by bowtie2. 

The bowtie2 index was built from FASTA files of each chromosome:`bowtie2-build 1.fa,...,6.fa,7_all_Ns.fa,8.fa,...`. The header of the BAM file had chr7 missing and a longer chr6. The new length chr6 was the sum of chr6 and chr7:

```
@SQ     SN:5    LN:...
@SQ     SN:6    LN:(chr6 length + chr7 length)
@SQ     SN:8   LN:...
```

We were wondering if this is the intentional behavior of bowtie2 and what's the recommended way of masking a whole chromosome for bowtie2 alignment.

Thanks for your help in advance.

**Comment by ccwang002** [comment]
This seems to be related to #192. Though in our case, we want to mask the entire chromosome and keep the original coordinates.

**Comment by ch4rr0** [comment]
The [commit](https://github.com/BenLangmead/bowtie2/commit/8fe6bb17984070319ab2dc64d6acc15f406f525d) referenced in issue #192 also addresses this issue. Below provides evidence of that.

Given the reference
```
>DUMMY_1
AAAGACTGCGCAATTTCTGAAAACTGAATTCGAATAAATTTAGGGTTGAAGTCATTTAAA
TTTCCTGTGAAGTTGTTGGTTCGTGTGTTGATAAGTGATGAGCTGATTGACTGATACGAA
TTGTTTTGACGCTTGCATCTTTAGACTTTTTTGTACGGGGTGTTGCAAGCCAGTTCTCAA
ACAATGCGTGCCAAGATCCAGCCAGTCAAAGCGATGTGACTCAGCGTCACGAGCACAGAT
GCACCGTGCAGCCATTTGAAACGCAGCCGGTCATCAGAGTCAGTGGCTCGATTGATTGCT
GGCATCAGCAACTGCCGCACGGGGATGGTCGTTAAAGCAACGGCCGCCATAACAATTGCA
GCCATGGTGTCGTGAGACCACCACAATAAAGCGGCAACGGCCGCCGTAGCCACCACAAAC
AAATAGAACCAAGGAAAAGCAAGCCGCAGCGTAGCACCAGCAGATGGGGCTGGAAGTGCT
TTGAATAAAAAAGCAGCAAAGCCAAAGGAGTACAGGGTCATGCCGCCGAACAACAGCGCG
GTGACCAATAT
>DUMMY_0
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
NNNNNNNNNNNNN
>DUMMY_2
TGGATAGGCAAAAAACAAAAAAGCCGCCTGGTGAACCAGGCGGCTTTGTTGGGGTGAATT
GACAGAACTTTTGTTCAAAGGCATAAAAACATTATGCACCAGCAACGTGACACCCTCGAA
CAGCGGTATCAAGCAGAAGTAGTGGCCTTTTTCAAATCCTTGACGGCTTGAAGAACTTCC
TCAACATGGCCTGGGACTTTGAGGCCTCTCCATTCGTTTTTGAGGGTGCCATCAGGGCCG
ACCAAAAATGTACTGCGCTCAATGCCTTTGACTTTCTTGCCGTACATGATTTTGTTTTTG
ACCACGCCAAACATGTGGCACATCTTTTCTTCCGTGTCCGCAATGAGTTCAAAGGGCAAC
TCAAGTTTGGCTTTGAAATCATCGTGGGACTTCATGTTATCGCGTGAGACACCCAAAACA
AGAGCACCAGCCTTGACGAAATCTTTGTATTTGTCGCGGAACTGCATGGCTTCTGTGGTG
CAACCCGGTGTGTTGTCTTTCGGGTAGAAAAACAAAACAACGGTTTGGCCGACGTGAGAG
ATGTTCGAAACTTTCAATCCGCTGGTGGCGTTGGCTTCAAATTCGGGGAGGGGTTTGTTC
AGAACGATCGCCATGGAACTTTAACTCGCTAGTCTGGTTGTGCGTACAGCACCTGCATGT
GGTGCCATTGCCGCTGGAGGCCAATAACCTCGGCAACCAGGTATTTTACTCTGAACGGCG
TTTTGACAGGCAACAACTGCCCA
```
Before the change Bowtie2 would output the following header. The length of `DUMMY_1`, 624, includes the length of the gap.

```
@HD     VN:1.0  SO:unsorted
@SQ     SN:DUMMY_1      LN:624
@SQ     SN:DUMMY_2      LN:743
```

After the commit the the length of `DUMMY_1` excludes the length of the gap.
```
@HD     VN:1.0  SO:unsorted
@SQ     SN:DUMMY_1      LN:551
@SQ     SN:DUMMY_2      LN:743
```

These changes were released as part of `v2.3.4.2`

Feel free to reopen this thread if this is still an issue.
